cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -401,9 +414,17 @@ runTransactionBuildEstimateCmd -- TODO change type
 
     ledgerPParams <-
       firstExceptT TxCmdProtocolParamsError $ readProtocolParameters sbe protocolParamsFile
-    inputsAndMaybeScriptWits <-
-      firstExceptT TxCmdScriptWitnessError $
-        readScriptWitnessFiles sbe txins
+    txInsAndMaybeScriptWits <-
+      mapM
+        ( \(txin, mSWit) -> do
+            case mSWit of
+              Nothing -> return (txin, Nothing)
+              Just sWit -> do
+                f <- firstExceptT TxCmdCliSpendingScriptWitnessError $ readSpendScriptWitness sbe sWit
+                return (txin, Just f)
+        )
```

Comment:
This curently is too verbose for what it does. I suggest something like:
```suggestion
      mapM
        ( \(txin, mSWit) -> (
            (txin,) <$> forM mSwit $
                  firstExceptT TxCmdCliSpendingScriptWitnessError . readSpendScriptWitness sbe
        )
```
or creating a specialised function e.g.
```haskell
forTxInsWitsM :: [(TxIn, Maybe (ScriptWitnessFiles WitCtxTxIn))]
              -> (ScriptWitnessFiles WitCtxTxIn -> m ScriptWitness WitCtxTxIn)
              -> m [(TxIn, Maybe (ScriptWitness WitCtxTxIn))]
```
since it's already done in 3 places